src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -948,15 +948,29 @@ bool Pane::_CanSplit(SplitState splitType)
 
     const Size minSize = _GetMinSize();
 
-    if (splitType == SplitState::Vertical)
+    auto actualSplitType = splitType;
+
+    // If the requested split type was "auto", determine which direction to
+    // split based on our current dimensions
+    if (actualSplitType == SplitState::Automatic)
+    {
+        actualSplitType = actualSize.Width >= actualSize.Height ? SplitState::Vertical : SplitState::Horizontal;

```

Comment:
careful taking a dep on actual size: this may fail for cmdline args since sizes dont exist on startup

Reply:
Added a note, but for whatever reason, the problems I had with panes having a 0 size on startup are _magically gone_. So at least for #4023, this'll work (when merged)